dawn/renderer/components/Editor.js

```diff
@@ -134,48 +155,56 @@ class Editor extends React.Component {
     setTimeout(() => this.refs.CodeEditor.editor.resize(), 0.1);
   }
 
-  sendCode(command) {
+  upload() {
+    const filepath = this.props.filepath;
+    if (filepath == null) {
+      console.log('No file? No upload');
+      return;
+    }
     const correctedText = this.correctText(this.props.editorCode);
     if (correctedText !== this.props.editorCode) {
       this.props.onAlertAdd(
         'Invalid characters detected',
         'Your code has non-ASCII characters, which won\'t work on the robot. ' +
-        'Please remove them and try again.'
+        'Please remove them and try again.',
       );
-      return false;
+      return;
     }
-    console.log('Deprecated: sendCode');
-    console.log(`Debug Info: ${command} EOD`);
-    return true;
-  }
-
-  upload() {
-    this.sendCode('upload');
+    const conn = new Client();
+    conn.on('ready', () => {
+      conn.sftp((err, sftp) => {
+        if (err) throw err;
+        console.log('SSH Connection');
+        sftp.fastPut(filepath, Editor.pathToName(filepath), (err2) => {
+          if (err2) throw err2;
+        });
+      });
+    }).connect({
+      debug: (inpt) => { console.log(inpt); },
+      host: this.props.ipAddress,
+      port: 22 });
+    setTimeout(() => { conn.end(); }, 2000);
   }
 
   startRobot() {
-    const sent = this.sendCode('execute');
-    if (sent) {
-      this.props.onClearConsole();
-    }
+    this.props.onUpdateCodeStatus(1);
+    this.props.onClearConsole();
   }
 
   stopRobot() {
-    console.log('Deprecated');
+    this.props.onUpdateCodeStatus(0);
   }
 
-  openAPI() {
-    window.open('https://pie-api.readthedocs.org/');
+  toggleUpdateModal() {
+    this.setState({ showUpdateModal: !this.state.showUpdateModal });
   }
 
-  pathToName(filepath) {
-    if (filepath !== null) {
-      if (process.platform === 'win32') {
-        return filepath.split('\\').pop();
-      }
-      return filepath.split('/').pop();
-    }
-    return '[ New File ]';
+  toggleConfigModal() {
+    this.setState({ showConfigModal: !this.state.showConfigModal });
+  }
+
+  openAPI() {
+    window.open('https://pie-api.readthedocs.org/');
```

Comment:
#96 If it's internet based, I'm going to need to know more further details. I was going to grab a pdf and add it into Dawn's static folders to load.